cmd/kpod/load.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"github.com/containers/storage"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+const (
+	dockerArchive = "docker-archive:"
+	oci           = "oci:"
+)
+
+type loadOptions struct {
+	oci   bool
+	input string
+	quiet bool
+}
+
+var (
+	loadFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "oci",
+			Usage: "Load image into oci",
+		},
+		cli.StringFlag{
+			Name:  "input, i",
+			Usage: "Read from archive file, instead of STDIN",
+			Value: "/dev/stdin",
+		},
+		cli.BoolFlag{
+			Name:  "quiet, q",
+			Usage: "Suppress the output",
+		},
+	}
+	loadDescription = "Loads the image from docker-archive by default or from oci if the oci flag is set."
+	loadCommand     = cli.Command{
+		Name:        "load",
+		Usage:       "load an image from docker archive or oci",
+		Description: loadDescription,
+		Flags:       loadFlags,
+		Action:      loadCmd,
+		ArgsUsage:   "",
+	}
+)
+
+func loadCmd(c *cli.Context) error {
+	store, err := getStore(c)
+	if err != nil {
+		return err
+	}
+
+	args := c.Args()
+	if len(args) > 0 {
+		return errors.New("too many arguments. Requires exactly 1")
+	}
+
+	oci := c.Bool("oci")
+	input := c.String("input")
+	quiet := c.Bool("quiet")
+
+	opts := loadOptions{
+		oci:   oci,
+		input: input,
```

Comment:
@umohnani8 has a version of this that creates a temporary file when this stdin is a pipe, and then hands that to the function call. Not sure if there is an interface that takes a reader rather then just a file.